corokafka/utils/corokafka_offset_manager.cpp

```diff
@@ -52,7 +59,14 @@ OffsetManager::OffsetManager(corokafka::ConsumerManager& consumerManager) :
 void OffsetManager::queryOffsetsFromBroker(const std::string& topic,
                                            TopicSettings& settings)
 {
-    //Create the assigned partitions for tracking
+    //Check if timeout needs to be overwritten
+    std::chrono::milliseconds consumerTimeout = _consumerManager.getTimeout(topic);
+    if ((_brokerTimeout == std::chrono::milliseconds(-1)) ||
+        (_brokerTimeout >= std::chrono::milliseconds(0))) {
+        //Overwrite timeout
+        _consumerManager.setTimeout(topic, _brokerTimeout);
```

Comment:
I know we intend to use this at startup, but what's going to happen if the consumer manager is active while we create the offset manager (say if the consumer manager isn't started as paused)? Messing with the timeout seems problematic. Could we instead add an optional timeout parameter to at least some functions, that could be used instead of the ConsumerManager timeout?

Reply:
Since we use CppKafka as underlying API, some calls did not expose a timeout parameter. We would need to give them a PR and once approved, use it here. Ideally yes, not mocking with the consumer-level timeout is best.